scvi/module/_mrdeconv.py

```diff
@@ -217,14 +226,21 @@ def loss(
         px_rate = generative_outputs["px_rate"]
         px_o = generative_outputs["px_o"]
         gamma = generative_outputs["gamma"]
+        v = generative_outputs["v"]
 
         reconst_loss = -NegativeBinomial(px_rate, logits=px_o).log_prob(x).sum(-1)
 
         # eta prior likelihood
         mean = torch.zeros_like(self.eta)
         scale = torch.ones_like(self.eta)
-        glo_neg_log_likelihood_prior = -Normal(mean, scale).log_prob(self.eta).sum()
-        glo_neg_log_likelihood_prior += torch.var(self.beta)
+        glo_neg_log_likelihood_prior = (
+            -1e-10 * Normal(mean, scale).log_prob(self.eta).sum()
+        )
+        glo_neg_log_likelihood_prior += 5.0 * torch.var(self.beta)
+
+        v_sparsity_loss = (
+            self.l1_sparsity * torch.mean(torch.exp(self.beta)) * torch.abs(v).mean(-1)
```

Comment:
I don't understand what is the motivation behind having mean(exp(beta) here

Reply:
We regularize var(beta) in the loss but not mean(beta). In the generative model, we only use exp(beta)*v. Without also exp(beta) the sparsity loss will lead to bigger beta and thereby smaller v to minimize it. softmax(v) or normalizing v should also work here.

Reply:
Number of cell types yes. However, the sum over v is variable and having smaller sum leads to lower l1 loss, while for the generative model this can be compensated by larger beta values. It introduces a strange dependency.I agree that the current solution is not great. I guess dividing by sum(v) is the best solution.

Reply:
By the way, softmax for v instead of softplus is worse in performance, so the model needs the flexibility that sum(v) is not fixed.

Reply:
I expected that, softmax is not appropriate here. You may want to take the mean over cell types as well (i.e., divide by the sum).

Reply:
Even though I don't like the effect of beta, this effect is low when using reasonable values (below 100) and I adjusted the code at this position. Keep in mind that it reduces the estimated counts for each gene by a fixed values when the betas are smaller.